binary/src/Pos/Binary/Class/Core.hs

```diff
@@ -82,6 +109,15 @@ matchSize requestedSize lbl actualSize =
   when (actualSize /= requestedSize) $
     cborError (lbl <> " failed the size check. Expected " <> show requestedSize <> ", found " <> show actualSize)
 
+-- | Compute encoded size of an integer.
```

Comment:
I think this code is only valid for unsigned values, but you're using it also for signed values below.

Reply:
Thanks, it looks like a bug was lingering here. Fixed.